### PR TITLE
Configure bf assets

### PR DESCRIPTION
**Pull request by stebl**
## Summary

Configures assets BF needs for authentication

## Motivation

The working assets are currently in `seasoned-test` and need to be moved into source control.  

## Details

1. I exported all the assets from `seasoned-test`
2. Then I tried to clean up the resulting mess and leave only what is strictly necessary.  

This caused some weird diffs.  Some unrelated files show up as "renamed" rather than a delete and create b/c they were so similar.  I'll leave some notes on the oddities I've caught

### CODE REVIEW

File: assets/clients/Auth0-Rules-Engine.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "is_token_endpoint_ip_header_trusted": false,
-  "name": "Seasoned (Test Application)",
+  "name": "Auth0-Rules-Engine",
```
Comment (stebl) [inline assets/clients/Auth0-Rules-Engine.json:3]: Not a rename, this is deleting and creating entirely separate "applications"

File: assets/clients/Brushfire-Authentication.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  "allowed_logout_urls": [],
+  "jwt_configuration": {
+    "alg": "RS256",
+    "lifetime_in_seconds": 36000,
+    "secret_encoded": false
+  },
+  "client_aliases": [],
+  "token_endpoint_auth_method": "client_secret_post",
+  "app_type": "non_interactive",
+  "grant_types": ["password", "http://auth0.com/oauth/grant-type/password-realm"],
```
Comment (stebl) [inline assets/clients/Brushfire-Authentication.json:27]: I think these are correct from [this doc](https://auth0.com/docs/api-auth/grant/password).  
If we're only using one of these grants, I think we should remove the unused one

File: assets/clients/Brushfire-Management.json [hunk 3]
```diff
@@ -1,12 +1,12 @@
 {
   "is_token_endpoint_ip_header_trusted": false,
-  "name": "Brushfire",
-  "callbacks": [],
+  "name": "Brushfire-Management",
```
Comment (stebl) [inline assets/clients/Brushfire-Management.json:3]: This is another delete and recreate.  The `Brushfire` app that was here was only a placeholder.

File: assets/connections/google-oauth2.json [hunk 4]
```diff
@@ -7,5 +7,5 @@
   "strategy": "google-oauth2",
   "name": "google-oauth2",
   "is_domain_connection": false,
-  "enabled_clients": ["auth0-deploy-cli-extension", "Crossroads", "Comedor", "Seasoned (Test Application)", "Brushfire"]
+  "enabled_clients": []
```
Comment (stebl) [inline assets/connections/google-oauth2.json:10]: Google-oauth2 is admittedly convenient for testing, but we're not using it in prod so I removed it.

File: assets/database-connections/Username-Password-Authentication/database.json [hunk 5]
```diff
@@ -4,13 +4,35 @@
...
+      "dictionary": []
+    },
+    "brute_force_protection": true,
+    "password_no_personal_info": {
+      "enable": false
+    },
+    "password_complexity_options": {
+      "min_length": 8
+    },
+    "enabledDatabaseCustomization": false
```
Comment (stebl) [inline assets/database-connections/Username-Password-Authentication/database.json:26]: @claudio-seasoned - I'm assuming these were the changes you made to make password strength backwards compatible, but it's possible some of these are remnants of testing and shouldn't be in the final config

File: assets/grants/Auth0-Rules-Engine (Management API).json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+  "client_id": "Auth0-Rules-Engine",
+  "audience": "https://##AUTH0_TENANT_NAME##/api/v2/",
+  "scope": ["read:users", "update:users", "read:roles"]
+}
```
Comment (stebl) [inline assets/grants/Auth0-Rules-Engine (Management API).json:5]: Exporting data from Auth0 creates grants in an obnoxious format that is env-specific.  

It by default includes the tenant name and doesn't do substitution.  Substitutions are only done on writes.    
It also includes the tenant name in the file name, which I manually renamed.  The file name isn't important to Auth0's final config.    

Something to keep an eye on and maybe add scripting around in the future.

File: assets/database-connections/Username-Password-Authentication/database.json (hunk 5 above)
Comment (claudio-seasoned) [reply assets/database-connections/Username-Password-Authentication/database.json:26]: Tha password policy should be only 4 characters and nothing else to be backwards compatible. No dictionary, history or personal information checking.

**Pull request merged**